quarkus.otel.traces.sampler.arg doesn't support String value (e.g. cannot configure "jaeger_remote" sampler)

### Describe the bug

The [Quarkus OpenTelemetry Configuration Reference](https://quarkus.io/guides/opentelemetry#configuration-reference) mentions that the `quarkus.*` properties match the ones from the [OpenTelemetry SDK Autoconfigure](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md).

In the AutoConfigure SDK [sampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler) configuration section, the `otel.traces.sampler.arg` property isn't explicitly restricted to having a number/double value.
In fact, in the OpenTelemetry Specification 1.25.0, section [General SDK Configuration](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration), the `OTEL_TRACES_SAMPLER_ARG` env var is described as possibly containing a comma separated list in case of the `jaeger_remote` and `parentbased_jaeger_remote` samplers.

But the [quarkus.otel.traces.sampler.arg](https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry_quarkus.otel.traces.sampler.arg) configuration property explicitly has a `double` type.

That means that a configuration of 
```
quarkus.otel.traces.sampler: "jaeger_remote"
quarkus.otel.traces.sampler.arg: "endpoint=http://hono-jaeger:14250"
```
isn't supported and leads to this exception on startup:
```
ERROR: Failed to start application (with profile [prod])
java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.otel.traces.sampler.arg with the config value "endpoint=http://hono-jaeger:14250" threw an Exception whilst being converted SRCFG00031: Expected a double value, got "endpoint=http://hono-jaeger:14250"
	at io.smallrye.config.ConfigMappingProvider.mapConfigurationInternal(ConfigMappingProvider.java:1003)
	at io.smallrye.config.ConfigMappingProvider.lambda$mapConfiguration$3(ConfigMappingProvider.java:959)
	at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:28)
	at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:959)
	at io.smallrye.config.ConfigMappings.mapConfiguration(ConfigMappings.java:91)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:630)
	at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
	at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
	... 13 more
```

### Expected behavior

`quarkus.otel.traces.sampler.arg` not being restricted to a `double` type.

### Actual behavior

Setting `quarkus.otel.traces.sampler.arg` to a String value leads to an exception on startup.

### How to Reproduce?

Use a configuration like
```
quarkus.otel.traces.sampler: "jaeger_remote"
quarkus.otel.traces.sampler.arg: "endpoint=http://hono-jaeger:14250"
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.2.6

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryLegacyConfigurationTest.java
+++ b/extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryLegacyConfigurationTest.java
@@ -57,7 +57,7 @@ class OpenTelemetryLegacyConfigurationTest {
         assertEquals(TRUE, oTelRuntimeConfig.traces().includeStaticResources());
         assertEquals("always_off", oTelBuildConfig.traces().sampler());
         assertTrue(oTelRuntimeConfig.traces().samplerArg().isPresent());
-        assertEquals(2.0d, oTelRuntimeConfig.traces().samplerArg().get());
+        assertEquals("2.0d", oTelRuntimeConfig.traces().samplerArg().get());
         assertEquals(FALSE, otlpExporterBuildConfig.enabled());
         assertTrue(otlpExporterRuntimeConfig.traces().legacyEndpoint().isPresent());
         assertTrue(otlpExporterRuntimeConfig.traces().headers().isPresent());

--- a/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java
+++ b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java
@@ -39,12 +39,16 @@ public interface TracesRuntimeConfig {
     Boolean includeStaticResources();
 
     /**
-     * An argument to the configured tracer if supported, for example a ratio.
+     * Sampler argument. Depends on the `quarkus.otel.traces.sampler` property.
+     * Fallbacks to the legacy property <code>quarkus.opentelemetry.tracer.sampler.ratio</code>.
      * <p>
-     * Fallbacks to the legacy property <code>quarkus.opentelemetry.tracer.sampler.ratio</code>
-     * or defaults to `1.0`.
+     * When setting the stock sampler to `traceidratio` or `parentbased_traceidratio` you need to set a `double` compatible
+     * value between `0.0d` and `1.0d`, like `0.01d` or `0.5d`. It is kept as a `String` to allow the flexible customisation of
+     * alternative samplers.
+     * <p>
+     * Defaults to `1.0d`.
      */
     @WithName("sampler.arg")
     @WithDefault("1.0d")
-    Optional<Double> samplerArg();
+    Optional<String> samplerArg();
 }

--- a/integration-tests/opentelemetry-jaeger-remote/pom.xml
+++ b/integration-tests/opentelemetry-jaeger-remote/pom.xml
@@ -1,0 +1,173 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>quarkus-integration-test-opentelemetry-jaeger-remote</artifactId>
+    <name>Quarkus - Integration Tests - Opentelemetry - Jaeger - Remote</name>
+
+    <properties>
+        <javax-ws-rs-api.version>2.1.1</javax-ws-rs-api.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry</artifactId>
+        </dependency>
+
+        <!-- JAX-RS -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+        </dependency>
+
+        <!-- Verify that the client can co-exist with the server -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
+        </dependency>
+
+        <!-- Test Dependencies -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.github.docker-java</groupId>
+            <artifactId>docker-java</artifactId>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.checkerframework</groupId>
+                    <artifactId>checker-qual</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.sun.activation</groupId>
+                    <artifactId>jakarta.activation</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>jcl-over-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>com.github.docker-java</groupId>
+            <artifactId>docker-java-transport-zerodep</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <!-- Minimal test dependencies to *-deployment artifacts for consistent build order -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-reactive-jackson-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-jackson-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-sdk-extension-jaeger-remote-sampler</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-exporter-jaeger</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>native-image</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
+</project>

--- a/integration-tests/opentelemetry-jaeger-remote/src/main/resources/META-INF.services/io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSamplerProvider
+++ b/integration-tests/opentelemetry-jaeger-remote/src/main/resources/META-INF.services/io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSamplerProvider
@@ -1,0 +1,1 @@
+io.opentelemetry.sdk.extension.trace.jaeger.sampler.JaegerRemoteSamplerProvider

--- a/integration-tests/opentelemetry-jaeger-remote/src/main/resources/META-INF.services/io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterProvider
+++ b/integration-tests/opentelemetry-jaeger-remote/src/main/resources/META-INF.services/io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterProvider
@@ -1,0 +1,1 @@
+io.opentelemetry.exporter.jaeger.internal.JaegerGrpcSpanExporterProvider

--- a/integration-tests/opentelemetry-jaeger-remote/src/main/resources/application.properties
+++ b/integration-tests/opentelemetry-jaeger-remote/src/main/resources/application.properties
@@ -1,0 +1,7 @@
+# Setting these for tests explicitly. Not required in normal application
+quarkus.application.name=opentelemetry-integration-test-jeager-remote
+quarkus.application.version=999-SNAPSHOT
+
+quarkus.otel.traces.sampler=jaeger_remote
+quarkus.otel.traces.sampler.arg=endpoint=http://localhost:14250,pollingInterval=5000,initialSamplingRate=1.0
+quarkus.otel.traces.exporter=jaeger

--- a/integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java
+++ b/integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java
@@ -1,0 +1,7 @@
+package io.quarkus.it.opentelemetry.jaeger;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class OTelJaegerRemoteIT extends OTelJaegerRemoteTest {
+}

--- a/integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
+++ b/integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
@@ -1,0 +1,116 @@
+package io.quarkus.it.opentelemetry.jaeger;
+
+import static com.github.dockerjava.api.model.HostConfig.newHostConfig;
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.time.Duration;
+
+import jakarta.inject.Inject;
+
+import org.awaitility.Awaitility;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.Ports;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.transport.DockerHttpClient;
+import com.github.dockerjava.zerodep.ZerodepDockerHttpClient;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Scope;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.path.json.JsonPath;
+
+@QuarkusTest
+public class OTelJaegerRemoteTest {
+
+    private static final int QUERY_PORT = 16686;
+    private static final int COLLECTOR_PORT = 14250;
+    private static final int HEALTH_PORT = 14269;
+    private static final String JAEGER_URL = "http://localhost";
+    private static final DockerClient dockerClient;
+    @Inject
+    OpenTelemetry openTelemetry;
+
+    static {
+        DockerClientConfig config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        DockerHttpClient httpClient = new ZerodepDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .build();
+        dockerClient = DockerClientImpl.getInstance(config, httpClient);
+        if (dockerClient.listContainersCmd().exec().stream()
+                .noneMatch(container -> container.getNames()[0].equals("/jaeger"))) {
+            ExposedPort queryPort = ExposedPort.tcp(QUERY_PORT);
+            ExposedPort collectorPort = ExposedPort.tcp(COLLECTOR_PORT);
+            ExposedPort hostPort = ExposedPort.tcp(HEALTH_PORT);
+            Ports portBindings = new Ports();
+            portBindings.bind(queryPort, Ports.Binding.bindPort(QUERY_PORT));
+            portBindings.bind(collectorPort, Ports.Binding.bindPort(COLLECTOR_PORT));
+            portBindings.bind(hostPort, Ports.Binding.bindPort(HEALTH_PORT));
+            CreateContainerResponse container = dockerClient
+                    .createContainerCmd("ghcr.io/open-telemetry/opentelemetry-java/jaeger:1.32")
+                    .withExposedPorts(queryPort, collectorPort, hostPort)
+                    .withHostConfig(newHostConfig().withPortBindings(portBindings))
+                    .withName("jaeger")
+                    .exec();
+            dockerClient.startContainerCmd(container.getId()).exec();
+        }
+    }
+
+    @AfterAll
+    static void teardown() {
+        dockerClient.listContainersCmd().exec()
+                .forEach(container -> {
+                    dockerClient.stopContainerCmd(container.getId()).exec();
+                    dockerClient.removeContainerCmd(container.getId()).exec();
+                });
+    }
+
+    @Test
+    void testJaegerRemoteIntegration() {
+        createTestSpan(openTelemetry);
+        JsonPath spanData = Awaitility.await()
+                .atMost(Duration.ofSeconds(30))
+                .until(this::assertJaegerHaveTrace, spanDataObj -> spanDataObj.get("data[0].spans[0].traceID") != null);
+        assertNotNull(spanData);
+        assertNotNull(spanData.get("data[0].spans[0].spanID"));
+        assertEquals("Test span", spanData.get("data[0].spans[0].operationName"));
+        assertEquals("Test event", spanData.get("data[0].spans[0].logs[0].fields[0].value"));
+        assertEquals("opentelemetry-integration-test-jeager-remote", spanData.get("data[0].processes.p1.serviceName"));
+    }
+
+    private void createTestSpan(OpenTelemetry openTelemetry) {
+        Span span = openTelemetry.getTracer(getClass().getCanonicalName()).spanBuilder("Test span").startSpan();
+        try (Scope scope = span.makeCurrent()) {
+            span.addEvent("Test event");
+        } catch (Throwable t) {
+            span.recordException(t);
+            throw t;
+        } finally {
+            span.end();
+        }
+    }
+
+    private JsonPath assertJaegerHaveTrace() {
+        String serviceName = ConfigProvider.getConfig().getConfigValue("quarkus.application.name").getValue();
+        String path = String.format("/api/traces?service=%s", serviceName);
+
+        return given()
+                .baseUri(String.format(JAEGER_URL + ":%d", QUERY_PORT))
+                .contentType("application/json")
+                .when().get(path)
+                .body()
+                .jsonPath();
+    }
+
+}

